From 1644a441ba5951475b9dbddff80e7d3d57a57ca4 Mon Sep 17 00:00:00 2001
From: ysiyer <yegnesh.s.iyer@intel.com>
Date: Fri, 20 Mar 2015 15:02:15 +0530
Subject: [PATCH] DC TI GPADC: Change suspend and resume to late suspend and
 early init

Issue: With the TI dollarcove PMIC, while system resumed from low
power state, the temperture read operation on pmic thermistors failed.
Root cause was the adc driver was in suspend state when the system
issued a temp read.

Solution: Changed the suspend and resume calls to late suspend and
early init, such that while the temp read is issued, the driver is
out from suspend state.

Change-Id: I7711d98b1cf438b3382e09aaa0e155a9c54ca0b0
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1248
Signed-off-by: ysiyer <yegnesh.s.iyer@intel.com>
---
 drivers/iio/adc/dc_ti_gpadc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/iio/adc/dc_ti_gpadc.c b/drivers/iio/adc/dc_ti_gpadc.c
index 2549bad..ddba310 100644
--- a/drivers/iio/adc/dc_ti_gpadc.c
+++ b/drivers/iio/adc/dc_ti_gpadc.c
@@ -470,8 +470,8 @@ static int dc_ti_gpadc_resume(struct device *dev)
 #endif
 
 static const struct dev_pm_ops dc_ti_gpadc_pm_ops = {
-	.suspend	= dc_ti_gpadc_suspend,
-	.resume		= dc_ti_gpadc_resume,
+	.suspend_late	= dc_ti_gpadc_suspend,
+	.resume_early	= dc_ti_gpadc_resume,
 };
 
 static struct platform_driver dc_ti_gpadc_driver = {
-- 
1.9.1

